frontend/src/metabase/query_builder/components/QueryModals.jsx

```diff
@@ -278,6 +279,10 @@ class QueryModals extends React.Component {
           onClose={onCloseModal}
         />
       </Modal>
+    ) : modal === MODAL_TYPES.PREVIEW_QUERY ? (
```

Comment:
This is one of the places I think converting an if...else into a switch case would make it a lot easier to glance.

Here's another example
https://github.com/metabase/metabase/blob/4394c2e0080c94a634ee05992bcd7a860d0d218f/frontend/src/metabase/visualizations/components/skeletons/ChartSkeleton/ChartSkeleton.tsx#L33-L43

What do you think?